### ISSUE TITLE
Sampling frames using pre-featurized dataset loader

### ISSUE THREAD

**Issue by kyleabeauchamp** [msmbuilder/msmbuilder#523]
So I used DihedralFeaturizer at the command line to build a tICA model of a very large dataset.  However, at this point I want to sample some frames.  Do people have a good idiom for this?

Do we have a good way to map trajectory frames through our command line pipeline?  My current understanding is that the explicitly globbed filenames never actually get saved in the provenance (see below), making the metadata somewhat incomplete.  

```
In [6]: print d.provenance 
MSMBuilder Dataset:
  MSMBuilder:	3.2.0.dev0.dev-dd4ea81
  Command:	/home/kyleb/opt/bin/msmb DihedralFeaturizer --trj ./trajectories/*.h5 --out dihedrals/
  Path:		dihedrals//
  Username:	kyleb
  Hostname:	kyleb-DX4870
  Date:		March 30, 2015 03:07 PM
  Comments:		

== Derived from ==
MDTraj dataset:
  path:		./trajectories/*.h5
  topology:	None
  stride:	1
  atom_indices	None

```

**Comment by kyleabeauchamp** [comment]
Here's another slightly related use case: sometimes I want to filter datasets based on some metadata features.  For example, I might have a folding@home dataset with run0 and run1.  Right now I'm manually featurizing these as separate numpy directories, but I imagine it might be possible to filter these things by saving more metadata and processing it via some sort of pipeline.  Not sure.

**Comment by mpharrigan** [comment]
I've been having similar needs, although from the API. I've been saving a pandas dataframe with all my metadata, making sure to match the indices (i.e. row index <--> dataset "key"). It works pretty well. Maybe we can formalize it by including some machinery in `dataset`

**Comment by mpharrigan** [comment]
You can see evidence of this workflow in my recent PR's :)

**Comment by kyleabeauchamp** [comment]
Will look into this, needs moar pandas

**Comment by kyleabeauchamp** [comment]
More comments on the dataset key stuff: do you have any thoughts on "fancy indexing".  For example, I wanted to filter out some trajectories.  I was hoping the following would work, but it seems like we'd need to do a bit more coding to make that happen.

```
runs = np.array(map(lambda x: "run0" in x, trajectories.glob_matches)).astype('int')
Xf0 = np.concatenate(tica.transform(d[runs == 0]))
```

**Comment by kyleabeauchamp** [comment]
(Obviously this can still be done with a list comprehension.)

**Comment by mpharrigan** [comment]
cc #539 

What do you think about the scheme I've sketched out in these gists
https://gist.github.com/mpharrigan/c2b68d8d015133f1e387

**Comment by cxhernandez** [comment]
@mpharrigan: I'm guessing this was resolved by #768?
